### PR TITLE
Assigning make_shared result to variables in test

### PR DESCRIPTION
**Pull request by brawner**
This PR addresses issue #961, where std::make_shared was marked as `[[nodiscard]]` in the latest MS VC buildtools. This PR only affects test files, which caused the warnings.

Example on most up-to-date buildtools (Windows) without this PR
[Windows without PR](https://citest.ros2.org/job/ci_windows/100/)
With this PR
[Build Status](https://citest.ros2.org/job/ci_windows/103/)

Tests on current ci.ros2.org
Mac
[Build Status](https://ci.ros2.org/job/ci_osx/7374/)
Linux
[Build Status](https://ci.ros2.org/job/ci_linux/9004/)
Windows
[Build Status](https://ci.ros2.org/job/ci_windows/8939/)

### CODE REVIEW

File: rclcpp/test/test_parameter_client.cpp [hunk 1]
```diff
@@ -69,7 +69,9 @@ TEST_F(TestParameterClient, async_construction_and_destruction) {
   {
     ASSERT_THROW(
     {
+      auto asynchronous_client =
```
Comment (dirk-thomas) [inline rclcpp/test/test_parameter_client.cpp]: Please indent the following line since it is wrapped.

Same below.

Comment (brawner) [reply rclcpp/test/test_parameter_client.cpp]: I get a uncrustify error when I do that. Any suggestion on how to resolve?
https://ci.ros2.org/job/ci_linux/9007/testReport/(root)/projectroot/uncrustify/

Comment (dirk-thomas) [reply rclcpp/test/test_parameter_client.cpp]: That is weird. Plenty of existing code indents wrapped lines already.

**Comment by brawner** [comment]
Note, this merged in the version that failed uncrustify. I was trying to find a more suitable style solution.

**Comment by dirk-thomas** [comment]
I didn't recheck the force commit and assumed it resolve the problem. Sorry.

Please mark your PR in a way that it is obvious that it is not ready to be merged yet.

**Comment by brawner** [comment]
Sorry, I only forced committed on the last one to sign-off. I guess I could
have used that opportunity to also revert the change. Making a new PR and
running a CI build.

**Comment by dirk-thomas** [comment]
I fixed the uncrustify warning with b25213a186add0ee93e270d08c9cb5601ae88066.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/windows-ci-builds-are-now-officially-containerized-and-virtualized-on-ci-ros2-org/13099/1

**Pull request merged**